xclim/indices/_hydrology.py

```diff
@@ -279,3 +283,120 @@ def melt_and_precip_max(
     out = agg.resample(time=freq).max(dim="time")
     out.attrs["units"] = snw.units
     return out
+
+
+@declare_units(q="[discharge]")
+def flow_index(q: xr.DataArray, p: float = 0.95, **indexer) -> xr.DataArray:
+    """
+    Flow index
+
+    Calculate the pth percentile of daily streamflow normalized by the median flow.
+
+    Parameters
+    ----------
+    q : xr.DataArray
+        Daily streamflow data.
+    p : float
+        Percentile for calculating the flow index, between 0 and 1. Default of 0.95 is for high flows.
+    indexer
+        Indexing parameters to perform a temporal subset of the data.
+        It accepts the same arguments as :py:func:`xclim.indices.generic.select_time`.
+
+    Returns
+    -------
+    xr.DataArray
+        Normalized Qp, which is the p th percentile of daily streamflow normalized by the median flow.
+
+    References
+    ----------
+    1. Addor, Nans & Nearing, Grey & Prieto, Cristina & Newman, A. & Le Vine, Nataliya & Clark, Martyn. (2018). A Ranking of Hydrological Signatures Based on Their Predictability in Space. Water Resources Research. 10.1029/2018WR022606.
+    2. Clausen, B., & Biggs, B. J. F. (2000). Flow variables for ecological studies in temperate streams: Groupings based on covariance. Journal of Hydrology, 237(3–4), 184–197. https://doi.org/10.1016/S0022-1694(00)00306-1
+
+    """
+    qt = select_time(q, **indexer)
+    qp = qt.quantile(p, dim="time")
+    q_median = qt.median(dim="time")
+    out = qp / q_median
+    out.attrs["units"] = "1"
+    return out
+
+
+@declare_units(q="[discharge]")
+def high_flow_frequency(
+    q: xr.DataArray, threshold_factor: int = 9, freq: str = "A-SEP", **indexer
```

Comment:
```suggestion
    q: xr.DataArray, threshold_factor: int = 9, freq: str = "YS-SEP", **indexer
```
Was the choice of a frequency anchored on the end date on purpose ? I think we never use that in xclim, I would suggest something anchored on the start date instead.

Also, "A" is deprecated in favor of "YE". "A-SEP" would be "YE-SEP".

Reply:
Just to be sure, "A-SEP" and "YS-SEP" are not equivalent. "A-SEP" goes from october to september while "YS-SEP" goes from september to august. My comment was about using "YS", feel free to chose the anchoring month as you feel is most appropriate for the indicator.

Also, the doc string still contains references to "A-SEP".

Reply:
Got it. idea was to pick water year as default. So, I will keep it as A-Sep

Reply:
Why not YS-OCT ?

Reply:
yea, same concept. So I will change it to YS-OCT and update the docstring